specs/daito/OpenSDSVMWareNBPDesign.md

```diff
@@ -0,0 +1,204 @@
+# OpenSDS VMWare NBP Design
+
+**Author(s)**: [zengyingzhe](https://github.com/zengyingzhe), [Sanil Kumar](https://github.com/skdwriting), [Najmudheen](https://github.com/NajmudheenCT), [Himanshu](https://github.com/himanshuvar)
+
+## Summary
+OpenSDS provides multiple north bound platforms support (viz., Kubernetes, OpenStack etc - Please refer the OpenSDS Architecture for more information-www.opensds.io). As a part of supporting more platforms, we plan to support VMWare platform as well.
+
+This documentation serves as the design spec for OpenSDS VMware NBP Architecture & Design.
+
+## Motivation
+
+VMware is one of the most important and widely used virtualization platforms, thus if we implement NBPs for VMware, it will greatly extend OpenSDS' usage scale(using in production environment) and also helps to improve the competitiveness of OpenSDS.
+
+### Goals
+* Support Specified VMware plugins and APIs using OpenSDS VMWare NBP (A tentative list is added below. However we need to further analyse on the feasibility and adaptation)
+* Build a OpenSDS VMWare NBP framework which can support multi-backend and single backend storage
+* Can support 3rd party backend adapters
+
+
+| NBP | Purpose |
+|--|--|
+| VASA(vSphere APIs for Storage Awareness) Provider | Provide the capabilities recognition of storage arrays to vSphere vCenter. |
+| VRO(vRealize Orchestrator) | Automate complex IT tasks for storage management. |
+| SRA(Storage Replication Adapter) | Cooperate with VMware SRM to provide the VMs protection by using array-based replication. |
+| vROps(vRealize Operations) | Provide integrated performance, capacity and configuration monitoring capabilities for storage arrays. |
+| vRLog(vRealize Log Insight) | Provide intuitive, actionable dashboards, sophisticated analytics for storage arrays. |
+
+Similar to the above plugins, we can support specific free standard vSphere Web Client plugins for storage management to ease the administrators for supported storage backends (already) in OpenSDS.
+
+
+### Non-Goals
+* No Data path plugins or features considered
+* Orchestration, Telemetry and Analytics specific plugins are low priority and may need to consider the current available features in OpenSDS for their design (consider based on the basic feasibility analysis)
+None.
+
+## High Level Architecture
+
+![](media/OpenSDS-VMWareNBP-Architecture-01.png)
```

Comment:
Formatting is wrong.  The diagram is not displayed.

Reply:
Actually the new images PR to merged to zen's branch. I have raised it.

Reply:
It is ok now.

Reply:
It seems that this diagram has an extra dotted line that should be removed.

Can you please delete the two old diagrams png files?  They are still part of the PR.